shell/platform/darwin/macos/framework/Source/FlutterEngineTest.mm

```diff
@@ -50,4 +52,66 @@
   [engine shutDownEngine];
 }
 
-}  // flutter::testing
+TEST(FlutterEngine, RegisterExternalTexture) {
+  FlutterEngine* engine = CreateTestEngine();
+  EXPECT_TRUE([engine runWithEntrypoint:@"main"]);
+
+  id<FlutterTexture> flutterTexture = OCMProtocolMock(@protocol(FlutterTexture));
+  bool called = false;
+
+  engine.embedderAPI.RegisterExternalTexture =
+      MOCK_ENGINE_PROC(RegisterExternalTexture, [&](auto engine, int64_t textureIdentifier) {
+        called = true;
+        EXPECT_EQ(textureIdentifier, reinterpret_cast<int64_t>(flutterTexture));
+        return kSuccess;
+      });
+
+  [engine.openGLRenderer registerTexture:flutterTexture];
```

Comment:
These tests are all testing the OpenGL renderer, so should be in a new test file for that class rather than going through the engine.